pkg/ir/deps/sql_program_deps_graph.go

```diff
@@ -0,0 +1,256 @@
+// Copyright 2020 The SQLFlow Authors. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package deps
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"sqlflow.org/sqlflow/pkg/parser"
+)
+
+// Statement represents a graph node of one statement.
+type Statement struct {
+	Statement string
+	// index of the statement in the SQL program
+	Order int
+	// Statement's input/output must be a table.
+	Inputs  []*Table
+	Outputs []*Table
+}
+
+// TableType can be table or model
+type TableType string
+
+const (
+	// TypeTable is a table node
+	TypeTable = "table"
+	// TypeModel is a TypeModel
+	TypeModel = "model"
+)
+
+// Table represents a graph node of one database table.
+type Table struct {
```

Comment:
How about `Node`, it's more meaningful in the graph concept.

Reply:
The current implementation is quite different from a general "graph", we return a list of statements with inputs and outputs, yet the input and output tables do not have it's input statements or output statements, this is simple compared to a general graph implementation.